tck/src/main/java/org/eclipse/microprofile/fault/tolerance/tck/asynchronous/AsyncClient.java

```diff
@@ -44,92 +40,21 @@
      * service 1 second in normal operation.
      *
      * @return the result as a Future
-     * @throws InterruptedException the interrupted exception
      */
     @Asynchronous
-    public Future<Connection> service() throws InterruptedException {
-
-        Connection conn = new Connection() {
-            {
-                Thread.sleep(1000);
-            }
-
-            @Override
-            public String getData() {
-                return "service DATA";
-            }
-        };
-
-        return CompletableFuture.completedFuture(conn);
+    public Future<Task> service(Task task) {
+        new AsyncBridge(task).perform(1000,"service DATA");
+        return CompletableFuture.completedFuture(task);
     }
 
     /**
-     * Service an operation until waitCondition is completed or 1 second timeout.
+     * service 1 second in normal operation.
      *
-     * @param waitCondition Execution of this method will delay until the condition is finished
-     * @param throwException Whether the method should throw an exception (true) or return a stage completed with exception (false)
-     * @return the result as a CompletionStage. It may be completed with
-     * InterruptedException if the thread is interrupted
+     * @return the result as a {@link CompletionStage}
      */
     @Asynchronous
-    public CompletionStage<Connection> serviceCS(Future<?> waitCondition, boolean throwException) {
-        return serviceCS(waitCondition, false, null);
-    }
-
-    @Asynchronous
-    public CompletionStage<Connection> serviceCS(Future<?> waitCondition) {
-        return serviceCS(waitCondition, false);
+    public CompletionStage<Task> serviceCS(Task task) {
+        new AsyncBridge(task).perform(1000,"service DATA");
+        return CompletableFuture.completedFuture(task);
```

Comment:
Similar the other async test client, the previous code was preferable. Avoid sleeps where possible.

Reply:
done